ui/js/controllers/perf/compare.js

```diff
@@ -121,63 +123,79 @@ perf.controller('CompareChooserCtrl', [
             };
 
             $scope.runCompare = function() {
-                ThResultSetModel.getResultSetsFromRevision($scope.originalProject.name, $scope.originalRevision).then(
-                    function() {
-                        $scope.originalRevisionError = undefined;
-                    },
-                    function(error) {
-                        $scope.originalRevisionError = error;
-                    }
-                );
+                var revisionPromise = [];
+                if ($scope.revisionComparison){
+                    revisionPromise.push.apply(revisionPromise, ThResultSetModel.getResultSetsFromRevision($scope.originalProject.name, $scope.originalRevision).then(
+                        function() {
+                            $scope.originalRevisionError = undefined;
+                        },
+                        function(error) {
+                            $scope.originalRevisionError = error;
+                        }
+                    ));
+                }
 
                 ThResultSetModel.getResultSetsFromRevision($scope.newProject.name, $scope.newRevision).then(
```

Comment:
You should be adding this to `revisionPromises`, not the code below.

I'd recommend reading the angular docs on promises: https://docs.angularjs.org/api/ng/service/$q